client/src/components/existing-request/existing-request-display.vue

```diff
@@ -279,11 +288,15 @@ export default class ExistingRequestDisplay extends Vue {
     newReqModule.mutateConditionsModalVisible(true)
   }
 
-  /** Open Incorporate Now Login Modal and apply NR Data to Session */
-  activateILModal () {
-    newReqModule.mutateIncorporateLoginModalVisible(true)
-    // nr persisted in the session to be used for affiliation/creation upon authentication in Signin.vue.
-    sessionStorage.setItem('NR_DATA', JSON.stringify(this.nr))
+  /** Affiliate the current Nr if authenticated or prompt login if unauthenticated */
+  affiliateOrLogin (): void {
+    if (this.isAuthenticated) {
+      this.createAffiliation(this.nr)
```

Comment:
Would be good to await this call (since it's async).

Is there a spinner or something while waiting for this to finish, since I imagine it takes several seconds to create the NR affiliation and to create the business.

Reply:
That's probably a good idea! Just out of curiosity though, with no subsequent actions and also since the requests within are awaited, does it matter to await this?

No spinner at the moment and there is no global state to disable the app. I guess using a spinner you could disable any further action until the requests are resolved, i'll look into this!

Reply:
In this case you do not have to await, since there is nothing after it that depends on it. (So you're throwing it out there to complete whenever.)

Yeah, you:
1. want the user to know something is happening while you create the affiliation and business
2. don't want the user to click something else if they get impatient

Reply:
Updated in next commit 👍